pyiron_base/project/generic.py

```diff
@@ -1514,6 +1513,33 @@ def unpack(self, origin_path, csv_file_name='export.csv', compress=True):
             self, archive_directory=origin_path, df=df, compressed=compress
         )
 
+    @classmethod
+    def register_tools(cls, name: str, creator):
```

Comment:
```suggestion
    def register_tools(cls, name: str, creator: Toolkit):
```

Reply:
No can do. I use `Toolkit` in the example, but right now I can't actually import it in the `project/generic` module or we get circular imports.

Also, if I understand what we want is `creator: typing.Type[Toolkit]` to hint that any child of `Toolkit` will do.

I wasn't willing to fight with circular imports just to get type hinting working :wink: